core/src/main/java/org/apache/carbondata/core/datastore/TableSpec.java

```diff
@@ -84,19 +84,20 @@ private void addDimensions(List<CarbonDimension> dimensions) {
     List<DimensionSpec> dictDimensionSpec = new ArrayList<>();
     int dimIndex = 0;
     DimensionSpec spec;
-    short actualPosition = 0;
+    short dictActualPosition = 0;
+    short noDimActualPosition = 0;
     // sort step's output is based on sort column order i.e sort columns data will be present
     // ahead of non sort columns, so table spec also need to add dimension spec in same manner
     for (int i = 0; i < dimensions.size(); i++) {
       CarbonDimension dimension = dimensions.get(i);
       if (dimension.isComplex()) {
-        spec = new DimensionSpec(ColumnType.COMPLEX, dimension, actualPosition++);
+        spec = new DimensionSpec(ColumnType.COMPLEX, dimension, (short) 0);
```

Comment:
Why is it '0' ?

Reply:
Here also please keep them in `noDimActualPosition`

Reply:
done